sdk/tables/azure-data-tables/azure/data/tables/_authentication.py

```diff
@@ -236,11 +236,13 @@ def _configure_credential(credential: None) -> None: ...
 
 
 def _configure_credential(
-    credential: Optional[Union[AzureNamedKeyCredential, AzureSasCredential, TokenCredential, SharedKeyCredentialPolicy]]
+    credential: Optional[Union[AzureNamedKeyCredential, AzureSasCredential, TokenCredential, SharedKeyCredentialPolicy]],
+    cosmos_endpoint: bool = False
 ) -> Optional[Union[BearerTokenChallengePolicy, AzureSasCredentialPolicy, SharedKeyCredentialPolicy]]:
     if hasattr(credential, "get_token"):
         credential = cast(TokenCredential, credential)
-        return BearerTokenChallengePolicy(credential, STORAGE_OAUTH_SCOPE)
+        scope = COSMOS_OAUTH_SCOPE if cosmos_endpoint else STORAGE_OAUTH_SCOPE
```

Comment:
We also need the equivalent in `azure.data.tables.aio._authentication_async.py`

Reply:
Resolved in latest commit